File: app/src/main/java/com/apps/adrcotfas/goodtime/Database/ProfileDao.java
package com.apps.adrcotfas.goodtime.Database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.apps.adrcotfas.goodtime.Profile;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface ProfileDao {

    @Insert(onConflict = REPLACE)
    void addProfile(Profile profile);

    @Query("select * from Profile")
    LiveData<List<Profile>> getProfiles();

    @Query("delete from Profile where name = :name")
    void deleteProfile(String name);
}

File: app/src/main/java/com/apps/adrcotfas/goodtime/database/ProfileDao.kt
package com.apps.adrcotfas.goodtime.database

import androidx.room.Dao
import androidx.room.OnConflictStrategy
import androidx.lifecycle.LiveData
import androidx.room.Insert
import androidx.room.Query

@Dao
interface ProfileDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addProfile(profile: Profile)

    @get:Query("select * from Profile")
    val profiles: LiveData<List<Profile>>

    @Query("delete from Profile where name = :name")
    fun deleteProfile(name: String)
}